app/src/main/res/layout/fragment_keyboard.xml

```diff
@@ -0,0 +1,255 @@
+        android:layout_marginStart="40dp"
+        android:layout_marginEnd="40dp"
+        android:background="@null"
+        android:gravity="center"
+        android:text="Select letters below to start typing."
+        android:textColor="@color/textColor"
+        android:textSize="48sp"
+        android:textStyle="bold"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:layout_marginTop="16dp"
+        android:orientation="vertical"
+        android:weightSum="4"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/keyboard_input">
+
+        <LinearLayout
+            android:id="@+id/predictive_text"
+            android:layout_width="match_parent"
+            android:layout_height="0dp"
+            android:layout_marginStart="40dp"
+            android:layout_marginEnd="40dp"
+            android:layout_weight="1"
+            android:background="@drawable/category_group_background"
+            android:gravity="center"
+            android:orientation="horizontal"
+            android:weightSum="5">
+
+            <com.willowtree.vocable.customviews.ActionButton
+                android:layout_width="0dp"
+                android:layout_height="104dp"
+                android:layout_weight="1"
+                android:background="@drawable/radio_button_default_background"
+                android:gravity="center" />
+
+            <com.willowtree.vocable.customviews.ActionButton
+                android:layout_width="0dp"
+                android:layout_height="104dp"
+                android:layout_weight="1"
+                android:background="@drawable/radio_button_default_background"
+                android:gravity="center" />
+
+            <com.willowtree.vocable.customviews.ActionButton
+                android:layout_width="0dp"
+                android:layout_height="104dp"
+                android:layout_weight="1"
+                android:background="@drawable/radio_button_default_background"
+                android:gravity="center" />
+
+            <com.willowtree.vocable.customviews.ActionButton
+                android:layout_width="0dp"
+                android:layout_height="104dp"
+                android:layout_weight="1"
+                android:background="@drawable/radio_button_default_background"
+                android:gravity="center" />
+
+            <com.willowtree.vocable.customviews.ActionButton
+                android:layout_width="0dp"
+                android:layout_height="104dp"
+                android:layout_weight="1"
+                android:background="@drawable/radio_button_default_background"
+                android:gravity="center" />
+
+        </LinearLayout>
+
+        <LinearLayout
```

Comment:
We are going to change this eventually to have single key layout. We may use this layout for a future "two-stroke" keyboard.

Reply:
When we re-visit this layout we can also add proper styles and pull out dimensions and strings.